app/lib/actions/katello/host/upload_profiles.rb

```diff
@@ -0,0 +1,60 @@
+module Actions
+  module Katello
+    module Host
+      class UploadProfiles < Actions::EntryAction
+        middleware.use Actions::Middleware::KeepCurrentUser
+
+        def plan(host, profile_string)
+          action_subject host
+
+          sequence do
+            plan_self(:host_id => host.id, :hostname => host.name, :profile_string => profile_string)
+            plan_action(GenerateApplicability, [host])
+          end
+        end
+
+        def humanized_name
+          if input.try(:[], :hostname)
+            _("Combined Profile Update for %s") % input[:hostname]
+          else
+            _('Combined Profile Update')
+          end
+        end
+
+        def resource_locks
+          :link
+        end
+
+        def rescue_strategy
+          Dynflow::Action::Rescue::Skip
+        end
+
+        def run
+          profiles = JSON.parse(input[:profile_string])
+          #free the huge string from the memory
+          input[:profile_string] = 'TRIMMED'.freeze
+          host = ::Host.find_by(:id => input[:host_id])
+          if host.nil?
+            Rails.logger.warn("Host with ID %s not found, continuing" % input[:host_id])
+          elsif host.content_facet.nil? || host.content_facet.uuid.nil?
+            Rails.logger.warn("Host with ID %s has no content facet, continuing" % input[:host_id])
+          else
+            package_profile = nil
+            enabled_repos = nil
+            profiles.each do |profile|
+              payload = profile["profile"]
+              case profile["content_type"]
+              when "rpm"
+                package_profile = payload
+              when "enabled_repos"
+                enabled_repos = payload
+              end
+            end
+            UploadPackageProfile.upload(input[:host_id], package_profile)
+            host.import_enabled_repositories(enabled_repos)
```

Comment:
I wonder if it makes more sense to do this and remove initializing the variables as nil above?
```suggestion
            host.import_enabled_repositories(enabled_repos) if defined?(enabled_repos)
```

Reply:
you don't even need defined?() there, they will be defined as nil
+1 to removing them

Reply:
to clarify, this would be sufficient:

```
host.import_enabled_repositories(enabled_repos) if enabled_repos
```

Reply:
true, I wasn't sure if @parthaa wanted to allow nil or some other falsey value for some reason since its passing nil now, but will let him make that judgement call

Reply:
You could do ` if defined?(package_profile) && package_profile` if you want to avoid passing in falsey values, but it looks like you could pass in nil before, I'm not sure if that was intentional.

Reply:
@johnpmitsch Removed the need for it. Look at my latest commit